### PR TITLE
Remove calls to Debug.WriteLine #61

### PR DESCRIPTION
**Pull request by kwhopper**
Attempt to address #61

### CODE REVIEW

File: MetadataExtractor/Formats/Exif/makernotes/OlympusMakernoteDirectory.cs [hunk 1]
```diff
@@ -441,16 +441,8 @@ private void ProcessCameraSettings([NotNull] byte[] bytes)
...
-            catch (IOException e)
-            {
-                // Should never happen, given that we check the length of the bytes beforehand.
-                Debug.WriteLine(e);
-            }
+            for (var i = 0; i < count; i++)
+                Set(CameraSettings.Offset + i, reader.GetInt32());
         }
 
         public bool IsIntervalMode()
```
Comment (kwhopper) [inline MetadataExtractor/Formats/Exif/makernotes/OlympusMakernoteDirectory.cs]: I agree based on the 'count' math above - this error should never happen

File: MetadataExtractor/Formats/Png/PngHeader.cs [hunk 2]
```diff
@@ -43,10 +43,7 @@ public PngHeader([NotNull] byte[] bytes)
             BitsPerSample = reader.GetByte();
 
             var colorTypeNumber = reader.GetByte();
-            var colorType = PngColorType.FromNumericValue(colorTypeNumber);
-            if (colorType == null)
-                throw new PngProcessingException("Unexpected PNG color type: " + colorTypeNumber);
-            ColorType = colorType;
+            ColorType = PngColorType.FromNumericValue(colorTypeNumber);
```
Comment (kwhopper) [inline MetadataExtractor/Formats/Png/PngHeader.cs]: With "Unknown" available, colorType will never be null

File: MetadataExtractor/Formats/Png/PngMetadataReader.cs [hunk 3]
```diff
@@ -102,14 +102,7 @@ public static
...
             {
-                try
-                {
-                    directories.AddRange(ProcessChunk(chunk));
-                }
-                catch (Exception e)
-                {
-                    Debug.WriteLine(e);
-                }
+                directories.AddRange(ProcessChunk(chunk));
```
Comment (kwhopper) [inline MetadataExtractor/Formats/Png/PngMetadataReader.cs]: There are various problems that can happen in ProcessChunk, so my vote is to let the exception happen and try to fix it afterwards.

File: MetadataExtractor/Formats/WebP/WebpRiffHandler.cs [hunk 4]
```diff
@@ -165,17 +190,23 @@ public void ProcessChunk(string fourCc, byte[] payload)
...
+                    {
+                        directory.Set(WebPDirectory.TagImageWidth, width);
+                        directory.Set(WebPDirectory.TagImageHeight, height);
                     }
+                    else
+                        directory.AddError(error);
+                    _directories.Add(directory);
                     break;
                 }
             }
```
Comment (kwhopper) [inline MetadataExtractor/Formats/WebP/WebpRiffHandler.cs]: For all of the cases in WebRiffHandler, the replacement goal was to always return a directory. If data isn't valid, those Sets are avoided but AddError will ensure the problem is propagated for possible review.

**Comment by kwhopper** [comment]
With "Unknown" available for Png, a few of the Png files from metatdata-extractor-images will emit more data instead of failing. This changes the images output a bit so it might be good to check over those and see if they seem appropriate. It's also possible other output may change with the WriteLine replacements.

File: MetadataExtractor/Formats/Exif/makernotes/OlympusMakernoteDirectory.cs (hunk 1 above)
Comment (drewnoakes) [inline MetadataExtractor/Formats/Exif/makernotes/OlympusMakernoteDirectory.cs]: Agree. This is likely a remnant from the Java version where checked exceptions can't determine this kind of thing.

File: MetadataExtractor/Formats/Exif/ExifReader.cs [hunk 5]
```diff
@@ -75,20 +75,21 @@ public sealed class ExifReader : IJpegSegmentMetadataReader
...
+                exifTiffHandler.Error("ExifReader TiffProcessingException : " + e.Message);
             }
             catch (IOException e)
             {
                 // TODO what do to with this error state?
-                Debug.WriteLine(e);
+                exifTiffHandler.Error("ExifReader IOException : " + e.Message);
             }
 
             return directories;
```
Comment (drewnoakes) [inline MetadataExtractor/Formats/Exif/ExifReader.cs]: Nice solution.

**Pull request merged**